org.eclipse.jdt.debug.jdi.tests/java19/TryVirtualThread.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * This is an implementation of an early-draft specification developed under the Java
+ * Community Process (JCP) and is made available for testing and evaluation purposes
+ * only. The code is not compatible with any specification of the JCP.
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+import java.io.OutputStream;
+import java.io.PrintWriter;
+
+public class TryVirtualThread implements Runnable {
+	/**
+	 * A <code>Thread</code> object
+	 */
+	public static Thread fThread;
+
+	/**
+	 * A <code>Thread</code> object representing the main thread
+	 */
+	public static Thread fMainThread;
+	
+	public static Thread fVirtualThread;
+
+	
+	/**
+	 * The instance of the <code>MainClass</code>
+	 */
+	public static TryVirtualThread fObject = new TryVirtualThread();
+
+	/**
+	 * An integer value
+	 */
+	public static int fInt = 0;
+	
+	/**
+	 * A string initialized to 'hello world'
+	 */
+	public static String fString = "Hello World";
+	
+	/**
+	 * The name of an event type
+	 */
+	public static String fEventType = "";
+	
+	/**
+	 * Runs the test program
+	 * @param args
+	 */
+	public static void main(java.lang.String[] args) {
+		// Ensure at least one carrier thread is created.
+		fVirtualThread = Thread.startVirtualThread(() -> {
+			System.out.println("Start a test virtual thread.");
+		});
+
+		ThreadGroup group = new ThreadGroup("Test ThreadGroup");
+		fThread = new Thread(group, fObject, "Test Thread");
+		fThread.start();
+
+		fMainThread = Thread.currentThread();
+
+		// Prevent this thread from dying
+		while (true) {
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException e) {
+			}
+		}
+	}
+	
+	@Override
+	public void run() {
+		try {
+			while (true) {
+				printAndSignal();
+				triggerEvent();
+				try {
+					Thread.sleep(1000);
+				} catch (InterruptedException e) {
+				}
+			}
+		} finally {
+			System.out.println("Running finally block in MainClass.run()");
+		}
+	}
+	
+	/**
+	 * Prints to System.out and throws an exception to indicate readiness
+	 */
+	synchronized public void printAndSignal() {
+		print(System.out);
+		// Signal readiness by throwing an exception
+		try {
+			throw new NegativeArraySizeException();
+		} catch (NegativeArraySizeException exc) {
+		}
+	}
+
+	public void print(OutputStream out) {
+		String string = fInt++ +". " + fString;
+		PrintWriter writer = new PrintWriter(out);
+		writer.println(string);
+		writer.flush();
+	}
+	
+	/**
+	 *	Trigger an event for the front-end.
+	 */
+	private void triggerEvent() {
+		/* Ensure we do it only once */
+		String eventType = fEventType;
```

Comment:
Where is fEventType being set for start and death? 
Did not understand the significance of this method.

Reply:
See [VirtualThreadTest.java](https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/76/files/e13b369cff7380730f1d19c953dc267cad6d3972#diff-4521fd6be4db9534ac9a358739142213a1f0675807934859baa9009044d3747e)#testJDIThreadEvents() on how to set ThreadStartEvent and ThreadDeathEvent.

![image](https://user-images.githubusercontent.com/14052197/183245871-b7ed51a2-79f1-4942-b745-a56a9de641b5.png)

Finally, it uses the JDI setValue method to set `fEventType` to the given eventType.

![image](https://user-images.githubusercontent.com/14052197/183245796-5627288f-023b-4c4e-acf3-e9c7f39668c4.png)

Reply:
Got it